### PR TITLE
Add vials

### PR DESCRIPTION
**Pull request by UbaserB**
## About the PR

I'm planning on atomising the **antagonist chemical kit** into two parts, one for vials, and the other for adding chemicals in these vials that syndicates can purchase. This is important for the future chemistry rework.

Vials are bottles that require a lot of damage to break and are a spawn round-start in chemistry lockers, but can also be obtained through cargo. They are also bigger than a bottle, as they are 1x2 In the future it would be cool to require reagents to be in vials instead of beakers/etc to be placed in the centrifuge, why they are much more expensive than bottles.

## Media

- [X] I have added screenshots/videos to this PR showcasing its changes ingame, **or** this PR does not require an ingame showcase

image
image

**Changelog**

:cl: Ubaser
- add: Chemistry now has vials, similar to bottles but are much more durable.

### CODE REVIEW

**Comment by deltanedas** [comment]
its abstract so either remove abstract or have a non-abstract prototype inherit it

**Comment by UbaserB** [comment]
> its abstract so either remove abstract or have a non-abstract prototype inherit it

what the hell does abstract mean

File: Resources/Prototypes/Entities/Objects/Specific/chemistry-vials.yml [hunk 1]
```diff
@@ -0,0 +1,92 @@
+# TODO: Have a seperate lid state that can be popped on and off (not as a seperate item) that changes whether the vial can be spilt or not
+
+- type: entity
+  name: vial
+  parent: BaseItem
+  id: BaseChemistryEmptyVial
+  abstract: true
```
Comment (TemporalOroboros) [inline Resources/Prototypes/Entities/Objects/Specific/chemistry-vials.yml:7]: This line marks the prototype as being primarily a template for other prototypes. If you remove it you should be able to spawn it in.

**Comment by JoeHammad1844** [comment]
so syndicate poison box will soon be real?

**Comment by UbaserB** [comment]
> so syndicate poison box will soon be real?

yessir

**Comment by UbaserB** [comment]
ok i finished the pr

File: Resources/Prototypes/Catalog/Fills/Boxes/medical.yml [hunk 2]
```diff
@@ -126,3 +126,21 @@
...
+  id: BoxVial
+  description: A box full of vials.
+  components:
+  - type: Storage
+    grid:
+    - 0,0,2,1
+  - type: StorageFill
+    contents:
+      - id: BaseChemistryEmptyVial
+        amount: 3
```
Comment (EmoGarbage404) [inline Resources/Prototypes/Catalog/Fills/Boxes/medical.yml:142]: make it 6 and also consider adding those little beaker stands. They'd just be whitelist storage that has itemmapper for displaying the fills.

**Review by EmoGarbage404 — CHANGES_REQUESTED** [review]

File: Resources/Prototypes/Entities/Objects/Specific/chemistry-vials.yml [hunk 3]
```diff
@@ -0,0 +1,95 @@
...
+      - state: vial-1-1
+        map: ["enum.SolutionContainerLayers.Fill"]
+        visible: false
+  - type: Appearance
+  - type: SolutionContainerVisuals
+    maxFillLevels: 6
+    fillBaseName: vial-1-
+  - type: SolutionContainerManager
+    solutions:
+      drink: # This solution name and target volume is hard-coded in ChemMasterComponent
```
[the remark this answers is not in the document]
Comment (UbaserB) [reply Resources/Prototypes/Entities/Objects/Specific/chemistry-vials.yml:31]: i just copy pasted it from another yaml

File: Resources/Prototypes/Entities/Objects/Specific/chemistry-vials.yml [hunk 4]
```diff
@@ -0,0 +1,95 @@
...
+    canChangeTransferAmount: true
+  - type: UserInterface
+    interfaces:
+    - key: enum.TransferAmountUiKey.Key
+      type: TransferAmountBoundUserInterface
+  - type: Item
+    size: Tiny
+    sprite: Objects/Specific/Chemistry/vial.rsi
+    shape:
+    - 0,0,0,1
```
[the remark this answers is not in the document]
Comment (UbaserB) [reply Resources/Prototypes/Entities/Objects/Specific/chemistry-vials.yml:53]: I thought it may help considering it is a buffed bottle, but okay

**Review by EmoGarbage404 — APPROVED** [review]

**Pull request merged**